sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+  ListSubscriptionsResult,
+  RuleResult,
+  ListRulesResult
+} from "./managementOperationInterfaces";
+
+/**
+ * All internal operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response:
+ *      {
+ *        isSuccessful: isSuccessful,
+ *        statusCode: statusCode,
+ *        body: body,
+ *        headers: headers
+ *      },
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ *
+ * Above information is used to construct and return a more strongly typed resource as `QueueResult`,
+ * `TopicResult`, `SubscriptionResult` or `RuleResult`.
+ *
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  private endpoint: string;
+
+  private queueResourceSerializer: AtomXmlSerializer;
+  private topicResourceSerializer: AtomXmlSerializer;
+  private subscriptionResourceSerializer: AtomXmlSerializer;
+  private ruleResourceSerializer: AtomXmlSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
+  }
+
+  /**
+   * Creates a queue.
+   * @param queuePath Name of the queue to use
+   * @param queueOptions
+   */
+  async createQueue(queuePath: string, queueOptions: QueueOptions): Promise<QueueResult> {
+    const response: HttpOperationResponse = await this._putResource(
+      queuePath,
+      queueOptions,
+      false,
+      this.queueResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Gets a queue.
```

Comment:
```suggestion
   * Returns an object representing the Queue with the given name along with all its properties
```

Please use the above format for docs for all the other get operations as well